api/logfilter/logfilter_test.go

```diff
@@ -69,88 +71,92 @@ var (
 		},
 	}
 
-	data1 = hash.Hash256b([]byte("topic1"))
-	data2 = hash.Hash256b([]byte("topic2"))
-	dataA = hash.Hash256b([]byte("topicA"))
-	dataB = hash.Hash256b([]byte("topicB"))
-	dataN = hash.Hash256b([]byte("topicNotExist"))
-
 	testData = []struct {
-		log   *action.Log
-		match [5]bool
-		exist [5]bool
+		log    *action.Log
+		match  [5]bool
+		exist  [5]bool
+		exist2 [5]bool
 	}{
 		{
 			&action.Log{
 				Address: "topicN",
-				Topics:  []hash.Hash256{dataN}, // both address and topic not exist
+				Topics:  []hash.Hash256{topicN}, // both address and topic not exist
 			},
 			[5]bool{true, false, false, false, false},
 			[5]bool{true, true, false, false, false},
+			[5]bool{true, false, false, false, false},
 		},
 		{
 			&action.Log{
 				Address: "topicN",
-				Topics:  []hash.Hash256{data1, data2, dataA}, // topic longer than log's topic list
+				Topics:  []hash.Hash256{topic1, topic2, topicA}, // topic longer than log's topic list
 			},
 			[5]bool{true, false, false, false, false},
 			[5]bool{true, true, true, true, true},
+			[5]bool{true, false, false, false, false},
 		},
 		{
 			&action.Log{
 				Address: "topicN",
-				Topics:  []hash.Hash256{data1, dataN}, // topic not match
+				Topics:  []hash.Hash256{topic1, topicN}, // topic not match
 			},
 			[5]bool{true, false, false, false, false},
 			[5]bool{true, true, true, true, false},
+			[5]bool{true, false, false, false, false},
 		},
 		{
 			&action.Log{
 				Address: "topic1",
-				Topics:  []hash.Hash256{dataN}, // topic not exist
+				Topics:  []hash.Hash256{topicN}, // topic not exist
 			},
 			[5]bool{true, true, false, false, false},
 			[5]bool{true, true, false, false, false},
+			[5]bool{true, true, false, false, false},
 		},
 		{
 			&action.Log{
 				Address: "topic2",
-				Topics:  []hash.Hash256{dataA, dataB}, // topic not match
+				Topics:  []hash.Hash256{topicA, topicB}, // topic not match
 			},
 			[5]bool{true, true, false, false, false},
 			[5]bool{true, true, true, false, true},
+			[5]bool{true, true, false, false, false},
 		},
 		{
 			&action.Log{
 				Address: "topicN",
-				Topics:  []hash.Hash256{data1, dataB},
+				Topics:  []hash.Hash256{topic1, topicB},
 			},
 			[5]bool{true, false, true, false, false},
 			[5]bool{true, true, true, true, true},
+			[5]bool{true, false, true, false, false},
```

Comment:
bfv2 exist check is more accurate than bfv1 (since address/topic sensitive information is added on)